types/protobuf.go

```diff
@@ -169,7 +169,14 @@ func (pb2tm) ValidatorSetFromProtoUpdate(
 			hasPublicKeys = false
 			break
 		}
+
+		pubkey, err := cryptoenc.PubKeyFromProto(*v.PubKey)
+		if err != nil || len(pubkey.Bytes()) == 0 {
```

Comment:
I wouldn't hide error if key is not valid

Reply:
done